### ISSUE TITLE
Question about parsing xml after changing the Historical Year

### ISSUE THREAD

**Issue by 837288251** [JGCRI/gcam-core#339]
I have changed the Historical year and have successfully generated the whole xml file. I have also got rid of the   "Historical Year" error checking in C++ source code. But when I what to run the run-gcam.bat file to parse xml file , the parsing work stopped after _socioeconomics_gSSP2.xml_ and let me press any key to close the cmd screen, withouting showing any further information. 
<img width="530" alt="264525388-3d3a8901-2348-41b2-bc11-132b42dea7b0" src="https://github.com/JGCRI/gcam-core/assets/90228576/e6fd3b2e-7d3d-41a2-8999-30811d97dbcc">

 I don't know if the problem is _socioeconomics_gSSP2.xml_ or _resources.xml_? Or the problem of share-weright?Thanks.

**Comment by abv666** [comment]
Did you solve it? I have the same problem

**Comment by 837288251** [comment]
> Did you solve it? I have the same problem

It seemed that there were some NAs in the xml, and technology disappearred under the subsector might also cause the silent crash.

**Comment by abv666** [comment]
Yes, I also found that there is a problem with some data in the xml file. How do you solve it?![Image](https://github.com/user-attachments/assets/6a4c1dc7-fafb-4fe8-bfbb-5da321033427)

**Comment by ssmithClimate** [comment]
I assume you are regenerating the set of XMLs using gcamdata. We are aware that the current R scripts are not robust to changing the historical years. We are working on this, and we are aiming for the next model release to have some significant improvements in this area.

**Comment by 837288251** [comment]
> Yes, I also found that there was a problem with some data in the xml file. How did you solve it?![Image](https://github.com/user-attachments/assets/6a4c1dc7-fafb-4fe8-bfbb-5da321033427)

It seems that you want to change the time step from 5 years to 1 year, which needs to make lots of changes to the gcamdata. As you showed in the picture, you need to find the csv file related to income-elasticity and fill the time gaps. However, I don't know whether the 1-year-time-step makes sense when you run gcam.bat,maybe it's related to the core C++ code.

**Comment by pkyle** [comment]
The issue is in `input/gcamdata/R/zenergy_L210.resources.R` which has the code
```
    A21.globalrsrctech_cost %>%
      repeat_add_columns(GCAM_region_names) %>%
      select(LEVEL2_DATA_NAMES[["ResReserveTechCost"]]) -> L210.ResTechCost

    A21.globalrsrctech_coef %>%
      repeat_add_columns(GCAM_region_names) %>%
      select(LEVEL2_DATA_NAMES[["ResReserveTechCoef"]])-> L210.ResTechCoef
```
That code block should be doing the standard interpolation/extrapolation to all model years. I copied an example block in a different issue https://github.com/JGCRI/gcam-core/issues/504#issuecomment-2572267958
Alternatively you could ad 1-year timesteps to the columns of data tables `energy/A21.globalrsrctech_coef.csv` and `energy/A21.globalrsrctech_cost.csv` but it would be a pain and you'd also need to update the "Column types" of those csv files to include all of the new years.
